### PR TITLE
Fix jit issue with relative epsilon and copying epsilons

### PR DESCRIPTION
**Pull request by JTT94**
Rearranged logic and changed `copy_epsilon` to not change-in-place.

https://colab.research.google.com/drive/1MYf0_rtuA8-mKuP1ZffrkWvIFJDBpUvh?usp=sharing

Clone updated repo
`!git clone https://github.com/JTT94/ott.git
`

Run test case with new code

`

    import os, sys
    
    sys.path.append('./ott') 
    
    import jax
    import jax.numpy as jnp
    import numpy as np
    import os, sys
    from ott.geometry import geometry
    from ott.geometry import pointcloud
    from ott.tools.sinkhorn_divergence import sinkhorn_divergence
    from functools import partial
    
    n, m, p, b = 5, 5, 10, 7
    rng = jax.random.PRNGKey(0)
    keys = jax.random.split(rng, 5)
    x = jax.random.normal(keys[0], (n, p))
    y = jax.random.normal(keys[1], (m, p)) + 1
    
    geom = pointcloud.PointCloud(x, y, power=2)
    
    @partial(jax.vmap, in_axes=[0, None, None, None])
    @partial(jax.vmap, in_axes=[None, 0, None, None])
    def sink_div(hist_1, hist_2, cost, epsilon):
        return sinkhorn_divergence(
            geometry.Geometry, cost, cost, cost, a=hist_1, b=hist_2, epsilon=epsilon
        ).divergence
    
    HIST_1 = jax.random.normal(keys[3], (b, n)) ** 2
    HIST_1 = HIST_1 / jnp.sum(HIST_1, axis=(1,), keepdims=True)
    
    HIST_2 = jax.random.normal(keys[4], (b, m)) ** 2
    HIST_2 = HIST_2 / jnp.sum(HIST_2, axis=(1,), keepdims=True)
    
    
    jax.jit(sink_div)(HIST_1, HIST_2, geom.cost_matrix, 0.01)

`

### CODE REVIEW

**Comment by marcocuturi** [comment]
Thanks James. Please split line 73 so that it is not longer than 80 characters
